clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -524,6 +524,58 @@ public void addMetric(MetricName metricName, MetricValueProvider<?> metricValueP
         addMetric(metricName, null, metricValueProvider);
     }
 
+    /**
```

Comment:
This makes sense, but I'm a bit curious about the motivation behind this change.

Reply:
@jolshan , the motivation is listed here =>
https://issues.apache.org/jira/projects/KAFKA/issues/KAFKA-13846?filter=allopenissues&orderby=created+DESC%2C+priority+DESC%2C+updated+DESC

As mentioned in the JIRA, currently the metric create operation relies on an IllegalArguementException in case of multi-threaded metric creation for the same metric. The other way is to syncrhonize the outer create call. This PR aims to make that process cleaner.